File: 5week/남규/5. Remove All Ones With Row and Column Flips.cpp
class Solution {
public:
    bool removeOnes(vector<vector<int>>& grid) {
        int N = grid.size();
        int M = grid[0].size();
        
        for(int col = 0; col < M; col++){
            if(grid[0][col]){
                for(int row = 1; row < N; row++){
                    grid[row][col] = grid[row][col] ^ 1;
                }
            }
        }
        
        for(int row = 1; row < N; row++){
            for(int col = 1; col < M; col++){
                if(grid[row][col - 1] != grid[row][col]){
                    return false;
                }
            }
        }
        return true;
    }
};

File: 5week/홍준/5.remove-all-ones-with-row-and-column-flips.py
## https://leetcode.com/problems/remove-all-ones-with-row-and-column-flips/

from typing import List

class Solution:
    def removeOnes(self, grid: List[List[int]]) -> bool:
        base = grid[0]
        opposite = [1-i for i in base]
        for row in grid:
            if row != base and row != opposite:
                return False
        return True